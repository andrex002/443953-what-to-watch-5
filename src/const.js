const filmsCount = {
  ALL: 30,
  PER_STEP: 8,
  SIMILAR: 4
};

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `Snatch`,
  `War of the Worlds`,
  `Mindhunter`,
  `Dardjeeling Limited`
];

const POSTERS = [
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `bohemian-rhapsody.jpg`,
  `macbeth.jpg`,
  `aviator.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
  `revenant.jpg`,
  `johnny-english.jpg`,
  `shutter-island.jpg`,
  `pulp-fiction.jpg`,
  `snatch.jpg`,
  `war-of-the-worlds.jpg`,
  `mindhunter.jpg`,
  `dardjeeling-limited.jpg`
];

const GENRES = [`Crime`, `Comedy`, `Drama`, `Documentary`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thriller`];

const DIRECTORS = [`James Cameron`, `Alfred Hitchcock`, `Steven Spielberg`];

const ACTORS = [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Bruce Lee`, `VanDamme`, `Jacke Chan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`];

const DESCRIPTION_TEXTS = [
  `It’s only after we’ve lost everything that we’re free to do anything.`,
  `It is not our abilities that show what we truly are… it is our choices.`,
  `Oh yes, the past can hurt. But you can either run from it, or learn from it.`,
  `Do, or do not. There is no “try”`,
  `Life moves pretty fast. If you don’t stop and look around once in a while, you could miss it.`,
  `I believe whatever doesn’t kill you, simply makes you…stranger.`,
  `The truth is a matter of circumstances, it’s not all things to all people all the time.`,
  `When I despair, I remember that all through history the way of truth and love has always won. There have been tyrants and murderers, and for a time, they can seem invincible, but in the end, they always fall. Think of it, always.`,
  `My momma always said, “Life is like a box of chocolates, you never know what you’re gonna get.`
];

const REVIEW_TEXTS = [
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
  `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`

];

export {filmsCount, TITLES, POSTERS, GENRES, DIRECTORS, ACTORS, DESCRIPTION_TEXTS, REVIEW_TEXTS};
