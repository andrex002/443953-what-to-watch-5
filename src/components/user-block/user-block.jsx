import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {AuthorizationStatus} from "../../const";

const UserBlock = (props) => {
  const {authorizationStatus, userName, userAvatar} = props;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH ?
        <div className="user-block__avatar">
          <Link to={`/mylist`}>
            <img src={userAvatar} alt={userName} width="63" height="63" />
          </Link>
        </div>
        : <Link to={`/login`} className="user-block__link">Sign in </Link>
      }
    </div>

  );
};

UserBlock.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  userName: PropTypes.oneOfType([PropTypes.string.isRequired, () => null]),
  userAvatar: PropTypes.oneOfType([PropTypes.string.isRequired, () => null]),
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
  userName: USER.name,
  userAvatar: USER.avatar
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
