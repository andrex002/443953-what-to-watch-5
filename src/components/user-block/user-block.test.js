import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {UserBlock} from "./user-block";
import {AuthorizationStatus} from "../../const";

describe(`Should UserBlock render correctly`, () => {
  it(`When user authorised`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UserBlock
          authorizationStatus={AuthorizationStatus.AUTH} />
          userName={`Alex`}
          userAvatar={`img/avatar.jpg`}
      </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When user not authorised`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UserBlock
          authorizationStatus={AuthorizationStatus.NO_AUTH} />
          userName={`Alex`}
          userAvatar={`img/avatar.jpg`}
      </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
