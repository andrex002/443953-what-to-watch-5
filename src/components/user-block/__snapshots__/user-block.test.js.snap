// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly When user authorised 1`] = `
Array [
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          height="63"
          width="63"
        />
      </a>
    </div>
  </div>,
  "userName=",
  "Alex",
  "userAvatar=",
  "img/avatar.jpg",
]
`;

exports[`Should UserBlock render correctly When user not authorised 1`] = `
Array [
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in 
    </a>
  </div>,
  "userName=",
  "Alex",
  "userAvatar=",
  "img/avatar.jpg",
]
`;
